packages/contracts/utils/contract.ts

```diff
@@ -42,13 +42,15 @@ export class AbstractContract {
    */
   public static async fromBuildArtifact(
     buildArtifact: BuildArtifact,
-    links?: { [name: string]: Promise<AbstractContract> }
+    links?: { [name: string]: Promise<AbstractContract> },
+    artifactName?: string
```

Comment:
If the argument will have a default value, it should be declared here:

```ts
artifactName: string = "UntitledContract"
```

This way you don't need to deal with the nuances of `T | undefined`.